diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -8,7 +8,7 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = src man tools doc include matlab python cmake examples
+SUBDIRS = src man tools doc include matlab cmake examples
 
 EXTRA_DIST = AUTHORS 00README.txt LICENSE.txt NEWS INSTALL \
 	Makefile.mk CMakeLists.txt windows maxima doc
@@ -42,4 +42,4 @@
 # install-matlab:
 # 	$(MAKE) -C matlab install-matlab
 
-.PHONY: doc install-doc man install-matlab install-python
+.PHONY: doc install-doc man install-matlab
