--- ./cmake/Python.cmake.orig	2009-09-09 00:55:19.000000000 +0300
+++ ./cmake/Python.cmake	2009-09-09 00:56:28.000000000 +0300
@@ -206,11 +206,11 @@
       MESSAGE(STATUS "Python bindings enabled")
     ELSE (SIP_IS_GOOD AND PYQT_IS_GOOD)
       SET (HAVE_PYTHON FALSE)
-      MESSAGE(STATUS "Python bindings disabled due dependency problems!")
+      MESSAGE(FATAL_ERROR  "Python bindings disabled due dependency problems!")
     ENDIF (SIP_IS_GOOD AND PYQT_IS_GOOD)
     
   ELSE (WITH_BINDINGS)
-    MESSAGE(STATUS "Python bindings disabled")
+    MESSAGE(SEND_ERROR "Python bindings disabled")
   ENDIF (WITH_BINDINGS)
 
 ENDIF (PYTHON_LIBRARIES AND PYTHON_INCLUDE_PATH)
