--- src/wimax.c.orig	2009-09-05 18:29:17.939340280 +0400
+++ src/wimax.c	2009-09-05 18:28:59.197340249 +0400
@@ -838,6 +838,20 @@
 	wmlog_msg(2, "Child exited with status %d", status);
 }
 
+void diode_switch(int sig)
+{
+  char req_data[0x400];
+  int len;
+
+  if (diode_on)
+    diode_on = 0;
+  else
+    diode_on = 1;
+
+  len = fill_diode_control_cmd(req_data, diode_on);
+  set_data(req_data, len);
+}
+
 int main(int argc, char **argv)
 {
 	struct sigaction sigact;
@@ -853,6 +867,8 @@
 	sigaction(SIGQUIT, &sigact, NULL);
 	sigact.sa_handler = sighandler_wait_child;
 	sigaction(SIGCHLD, &sigact, NULL);
+	sigact.sa_handler = diode_switch;
+	sigaction(SIGUSR1, &sigact, NULL);
 
 	if (logfile != NULL) {
 		set_wmlogger(argv[0], WMLOGGER_FILE, logfile);
