diff -r 3ed0a8b9bdc6 configure.ac
--- a/configure.ac	Tue Apr 13 05:36:25 2010 +0300
+++ b/configure.ac	Tue Apr 13 05:50:15 2010 +0300
@@ -224,35 +224,26 @@
 fi
 AM_CONDITIONAL([WITH_MYSQL], test "xyes" = "x$mysql")
 
-postgresql="yes"
+postgresql="no"
 AC_MSG_CHECKING(for postgresql)
 AC_ARG_WITH([postgresql], 
         AS_HELP_STRING([--with-postgresql(=<path>)], 
                 [path is optional and if given should specify the full path to the PostgreSQL
                  configure script, pg_config. E.g. --with-postgresql=/<path>/pg_config]), 
         [
-                if test "xno" = "x$with_postgresql"; then
-                        AC_MSG_RESULT([no])
-                        postgresql="no"
-                else
+                if test "xyes" = "x$with_postgresql"; then
                         AC_MSG_RESULT([yes])
-                        AC_CHECK_FILE([$with_postgresql], 
-                        [
-                                PGCONFIG=$with_postgresql
-                        ], 
-                        [
-                                AC_MSG_WARN([pg_config given as $with_postgresql not found])
-                                postgresql="no"
-                        ])        
+                     AC_PATH_PROG([PGCONFIG], [pg_config], [no], [$PATH:/usr/local/bin:/usr/bin])
+			if test "x$PGCONFIG" = "xno"; then
+				AC_MSG_WARN([pg_config is required to build libzdb with postgresql])
+				postgresql="no"
+			else
+				postgresql="yes"
+               		fi
                 fi
         ],
         [
-                AC_MSG_RESULT([yes])
-                AC_PATH_PROG([PGCONFIG], [pg_config], [no], [$PATH:/usr/local/bin:/usr/local/pgsql/bin])
-                if test "x$PGCONFIG" = "xno"; then
-                        AC_MSG_WARN([pg_config is required to build libzdb with postgresql])
                         postgresql="no"
-                fi
         ])
 if test "xyes" = "x$postgresql"; then
         svd_CPPFLAGS=$CPPFLAGS
