diff -r e623fb667991 configure.ac
--- a/configure.ac	Tue Apr 13 05:53:09 2010 +0300
+++ b/configure.ac	Tue Apr 13 08:20:25 2010 +0300
@@ -137,6 +137,7 @@
                 and SQLite is used from a multi-threaded program.]),
     [
         if test "x$enableval" = "xyes" ; then
+		PKG_CHECK_MODULES([sqlite3], [ sqlite3 >= 3.6.12 ])
 		SQLITEUNLOCK=1
                 CFLAGS="$CFLAGS -DSQLITEUNLOCK"
 	else
@@ -262,41 +263,37 @@
 fi
 AM_CONDITIONAL([WITH_POSTGRESQL], test "xyes" = "x$postgresql")
 
-sqlite="yes"
+sqlite="no"
 AC_MSG_CHECKING(for SQLite3)
 AC_ARG_WITH([sqlite], 
         AS_HELP_STRING([--with-sqlite=<path>], 
                 [prefix of SQLite installation. e.g. /usr/local/sqlite3 or /usr]), 
         [
-        if test "xno" = "x$with_sqlite"; then
-                AC_MSG_RESULT([no])
-                sqlite="no"
-        else
-                AC_MSG_RESULT([yes])
+        if test "x$with_sqlite" != "xno"; then
+		
+		PKG_CHECK_MODULES([sqlite3], [ sqlite3 ])
                 svd_LDFLAGS=$LDFLAGS
                 svd_CPPFLAGS=$CPPFLAGS
                 LDFLAGS="-L$with_sqlite/lib $LDFLAGS"
                 CPPFLAGS="-I$with_sqlite/include $CPPFLAGS"
                 AC_CHECK_HEADERS([sqlite3.h], [
-                        sqlite="yes"
-                        if test -r "$with_sqlite/lib/libsqlite3.a"; then
-                                DBCPPFLAGS="$DBCPPFLAGS -I$with_sqlite/include"
-                                DBLDFLAGS="$DBLDFLAGS -L$with_sqlite/lib/ -lsqlite3"
-                        else
-                                sqlite="no"
-                        fi
-                ], [sqlite="no"])
+			sqlite="yes"
+			DBCPPFLAGS="$DBCPPFLAGS `$PKG_CONFIG --cflags-only-I sqlite3`"
+			DBLDFLAGS="$DBLDFLAGS `$PKG_CONFIG  --libs-only-l sqlite3`"
+			], 
+			[sqlite="no"])
                 LDFLAGS=$svd_LDFLAGS
                 CPPFLAGS=$svd_CPPFLAGS
         fi
         ], 
         [
-                AC_MSG_RESULT([yes])
-                AC_CHECK_LIB([sqlite3], [sqlite3_open], [], [sqlite="no"])
+                sqlite="no"
         ])
+
 if test "xyes" = "x$sqlite"; then
-        AC_DEFINE([HAVE_LIBSQLITE3])
+        AC_DEFINE([HAVE_LIBSQLITE3], 1,  [Define to 1 to enable SQLite3])
 fi
+
 AM_CONDITIONAL([WITH_SQLITE], test "xyes" = "x$sqlite")
 
 AC_SUBST(DBLDFLAGS)
@@ -385,6 +382,8 @@
 fi
 if test "xtrue" = "x$openssl"; then
 echo "|   Openssl:                                      ENABLED    |"
+else
+echo "|   Openssl:                                      DISABLED    |"
 fi
 echo "|------------------------------------------------------------|"
 if test "xyes" = "x$sqlite"; then
