--- Makefile.in.orig	2009-08-29 21:03:07.000000000 +0300
+++ Makefile.in	2009-08-29 21:03:18.000000000 +0300
@@ -932,12 +932,6 @@
 	touch $(DESTDIR)/$(pamdir)/fingerprint-auth-ac
 	touch $(DESTDIR)/$(pamdir)/smartcard-auth-ac
 	mkdir -p $(DESTDIR)/$(datadir)/applications
-	desktop-file-install --vendor redhat \
-		--add-category=X-Red-Hat-Base \
-		--add-category=Application \
-		--add-category=System \
-		--dir=$(DESTDIR)/$(datadir)/applications \
-		authconfig.desktop
 
 uninstall-local:
 	$(RM) $(DESTDIR)/$(bindir)/authconfig
--- Makefile.am.orig	2009-08-29 21:14:20.000000000 +0300
+++ Makefile.am	2009-08-29 21:15:30.000000000 +0300
@@ -54,12 +54,6 @@
 	touch $(DESTDIR)/$(pamdir)/fingerprint-auth-ac
 	touch $(DESTDIR)/$(pamdir)/smartcard-auth-ac
 	mkdir -p $(DESTDIR)/$(datadir)/applications
-	desktop-file-install --vendor redhat \
-		--add-category=X-Red-Hat-Base \
-		--add-category=Application \
-		--add-category=System \
-		--dir=$(DESTDIR)/$(datadir)/applications \
-		authconfig.desktop
 
 uninstall-local:
 	$(RM) $(DESTDIR)/$(bindir)/authconfig
@@ -101,7 +95,6 @@
 DISTCLEANFILES = intltool-extract intltool-merge intltool-update authconfig.desktop
 
 VERSION=@PACKAGE_VERSION@
-TAG=$(shell echo r-$(VERSION) | sed 's,\.,-,g')
 distdir=$(PACKAGE)-$(VERSION)
 
 authconfig.glade.h: authconfig.glade
@@ -110,21 +103,5 @@
 update-po: authconfig.glade.h
 	$(MAKE) -C po update-po
 
-tag:
-	hg tag $(TAG)
-
-force-tag forcetag:
-	hg tag --force $(TAG)
-
 authconfig.spec: authconfig.spec.in configure.in
 	./autogen.sh
-
-archive:
-	rm -fr /tmp/$(PACKAGE) $(PACKAGE)-$(VERSION).tar.bz2
-	mkdir /tmp/$(PACKAGE) && \
-	hg archive -r $(TAG) /tmp/$(PACKAGE)/$(distdir) && \
-	(cd /tmp/$(PACKAGE)/$(distdir) && \
-	./autogen.sh && \
-	./configure && \
-	make dist) && \
-	mv /tmp/$(PACKAGE)/$(distdir)/$(distdir).tar.bz2 .
