diff --git a/qt4/style/qtcurve.cpp b/qt4/style/qtcurve.cpp
index 8bc70cd..b4d737f 100644
--- a/qt4/style/qtcurve.cpp
+++ b/qt4/style/qtcurve.cpp
@@ -123,25 +123,6 @@
 // TODO! REMOVE THIS WHEN KDE'S ICON SETTINGS ACTUALLY WORK!!!
 #define FIX_DISABLED_ICONS
 
-extern QString (*qt_filedialog_existing_directory_hook)(
-    QWidget *parent, const QString &caption, const QString &dir,
-    QFileDialog::Options options);
-
-extern QString (*qt_filedialog_open_filename_hook)(
-    QWidget *parent, const QString &caption, const QString &dir,
-    const QString &filter, QString *selectedFilter,
-    QFileDialog::Options options);
-
-extern QStringList (*qt_filedialog_open_filenames_hook)(
-    QWidget * parent, const QString &caption, const QString &dir,
-    const QString &filter, QString *selectedFilter,
-    QFileDialog::Options options);
-
-extern QString (*qt_filedialog_save_filename_hook)(
-    QWidget *parent, const QString &caption, const QString &dir,
-    const QString &filter, QString *selectedFilter,
-    QFileDialog::Options options);
-
 namespace QtCurve {
 
 static inline void
@@ -1399,13 +1380,6 @@ void Style::polish(QApplication *app)
         opts.menuBgndAppearance=APPEARANCE_FLAT;
     }
 
-    if (opts.useQtFileDialogApps.contains(appName)) {
-        qt_filedialog_existing_directory_hook = 0L;
-        qt_filedialog_open_filename_hook = 0L;
-        qt_filedialog_open_filenames_hook = 0L;
-        qt_filedialog_save_filename_hook = 0L;
-    }
-
     BASE_STYLE::polish(app);
     if (opts.hideShortcutUnderline) {
         app->installEventFilter(itsShortcutHandler);
diff --git a/qt5/style/qtcurve_api.cpp b/qt5/style/qtcurve_api.cpp
index d724241..5495a62 100644
--- a/qt5/style/qtcurve_api.cpp
+++ b/qt5/style/qtcurve_api.cpp
@@ -69,25 +69,6 @@
 
 #include <QDebug>
 
-extern QString (*qt_filedialog_existing_directory_hook)(
-    QWidget *parent, const QString &caption, const QString &dir,
-    QFileDialog::Options options);
-
-extern QString (*qt_filedialog_open_filename_hook)(
-    QWidget *parent, const QString &caption, const QString &dir,
-    const QString &filter, QString *selectedFilter,
-    QFileDialog::Options options);
-
-extern QStringList (*qt_filedialog_open_filenames_hook)(
-    QWidget * parent, const QString &caption, const QString &dir,
-    const QString &filter, QString *selectedFilter,
-    QFileDialog::Options options);
-
-extern QString (*qt_filedialog_save_filename_hook)(
-    QWidget *parent, const QString &caption, const QString &dir,
-    const QString &filter, QString *selectedFilter,
-    QFileDialog::Options options);
-
 namespace QtCurve {
 
 void
@@ -162,13 +143,6 @@ Style::polish(QApplication *app)
         opts.menuBgndAppearance = APPEARANCE_FLAT;
     }
 
-    if (opts.useQtFileDialogApps.contains(appName)) {
-        qt_filedialog_existing_directory_hook = 0L;
-        qt_filedialog_open_filename_hook = 0L;
-        qt_filedialog_open_filenames_hook = 0L;
-        qt_filedialog_save_filename_hook = 0L;
-    }
-
     BASE_STYLE::polish(app);
     if (opts.hideShortcutUnderline) {
         app->installEventFilter(itsShortcutHandler);
