--- m4/mozldap.m4.orig	2009-10-05 01:28:00.612188911 +0300
+++ m4/mozldap.m4	2009-10-05 01:28:41.360183594 +0300
@@ -27,7 +27,7 @@
   if test "$withval" = yes
   then
     AC_MSG_RESULT([using system MozLDAP])
-  elif test -e "$withval"/include/ldap.h -a -d "$withval"/lib
+  elif test -e "$withval"/include/mozldap/ldap.h -a -d "$withval"/lib
   then
     AC_MSG_RESULT([using $withval])
     LDAPSDKDIR=$withval


--- m4/nspr.m4.orig	2009-10-05 01:33:52.644184372 +0300
+++ m4/nspr.m4	2009-10-05 01:34:16.272180912 +0300
@@ -81,11 +81,6 @@
       nspr_lib=`$PKG_CONFIG --libs-only-L nspr`
       nspr_libdir=`$PKG_CONFIG --libs-only-L nspr | sed -e s/-L// | sed -e s/\ .*$//`
       AC_MSG_RESULT([using system NSPR])
-    elif $PKG_CONFIG --exists dirsec-nspr; then
-      nspr_inc=`$PKG_CONFIG --cflags-only-I dirsec-nspr`
-      nspr_lib=`$PKG_CONFIG --libs-only-L dirsec-nspr`
-      nspr_libdir=`$PKG_CONFIG --libs-only-L dirsec-nspr | sed -e s/-L// | sed -e s/\ .*$//`
-      AC_MSG_RESULT([using system dirsec NSPR])
     else
       AC_MSG_ERROR([NSPR not found, specify with --with-nspr.])
     fi


--- configure.ac.orig	2009-10-05 01:49:24.864183179 +0300
+++ configure.ac	2009-10-05 01:53:02.284184209 +0300
@@ -2,7 +2,7 @@
 # Process this file with autoconf to produce a configure script.
 AC_PREREQ(2.59)
 # This version is the version returned by ns-slapd -v
-AC_INIT([dirsrv], [1.2.2], [http://bugzilla.redhat.com/])
+AC_INIT([dirsrv], [1.2.2], [http://code.google.com/p/rion-overlay/issues/list/])
 # AC_CONFIG_HEADER must be called right after AC_INIT.
 AC_CONFIG_HEADERS([config.h])
 AM_INIT_AUTOMAKE([1.9 foreign subdir-objects])
@@ -186,8 +186,8 @@
 AC_PREFIX_DEFAULT([/opt/$PACKAGE_NAME])
 
 brand=389
-capbrand=389
-vendor="389 Project"
+capbrand=389-gentoo
+vendor="389 Project/Gentoo"
 
 m4_include(m4/fhs.m4)
 
@@ -258,8 +258,8 @@
 schemadir=/$PACKAGE_NAME/schema
 
 # default user, group
-defaultuser=nobody
-defaultgroup=nobody
+defaultuser=dirsrv
+defaultgroup=dirsrv
 
 AC_SUBST(configdir)
 AC_SUBST(sampledatadir)
