diff --git a/pki-native-tools-1.2.0/Makefile.am b/pki-native-tools-1.2.0/Makefile.am
--- a/pki-native-tools-1.2.0/Makefile.am
+++ b/pki-native-tools-1.2.0/Makefile.am
@@ -60,10 +60,10 @@
 #------------------------
 # Install Paths
 #------------------------
-confdir = $(prefix)@confdir@
-readmedir = $(prefix)@readmedir@
-samplesdir = $(prefix)@samplesdir@
-templatesdir = $(prefix)@templatesdir@
+confdir = @confdir@
+readmedir = @readmedir@
+samplesdir = @samplesdir@
+templatesdir = @templatesdir@
 
 #------------------------
 # Build Products
diff --git a/pki-native-tools-1.2.0/configure.ac b/pki-native-tools-1.2.0/configure.ac
--- a/pki-native-tools-1.2.0/configure.ac
+++ b/pki-native-tools-1.2.0/configure.ac
@@ -99,14 +99,16 @@
 # use /usr as the prefix for everything, which means
 # /usr/etc, /usr/opt, and /usr/var.  FHS sez to use
 # /etc, /opt, and /var.
-ac_default_prefix=/opt/native-tools
+ac_default_prefix=/usr
 prefix=$ac_default_prefix
 exec_prefix=$prefix
 # relative to prefix
-confdir=/conf
-readmedir=/doc
-samplesdir=/samples
-templatesdir=/templates
+sysconfigdir='/etc'
+confdir=$sysconfigdir
+datadir=$ac_default_prefix/share
+readmedir=$datadir/doc/$PACKAGE_NAME/
+samplesdir=$datadir/$PACKAGE_NAME/
+templatesdir=$datadir/$PACKAGE_NAME/templates
 AC_SUBST(confdir)
 AC_SUBST(readmedir)
 AC_SUBST(samplesdir)
