Index: laptop.h
===================================================================
--- laptop.h	(Revision 280)
+++ laptop.h	(Arbeitskopie)
@@ -637,6 +637,15 @@
 	},
 	{
 		.callback = dmi_matched,
+		.ident = "Toshiba Satellite A110",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Satellite A110"),
+		},
+		.driver_data = (void*) TSM70
+	},
+	{
+		.callback = dmi_matched,
 		.ident = "Toshiba Satellite P100",
 		.matches = {
 			DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
