diff -ur omnibook-9999.orig/Makefile omnibook-9999/Makefile
--- omnibook-9999.orig/Makefile	2011-10-28 23:33:07.000000000 +0600
+++ omnibook-9999/Makefile	2011-10-28 23:36:42.000000000 +0600
@@ -125,13 +125,7 @@
 
 ifeq ($(OMNIBOOK_WANT_BACKLIGHT),y)
 ifdef CONFIG_BACKLIGHT_CLASS_DEVICE
-# we support backlight interface only after 2.6.16
-ifeq ($(shell if [ $(SUBLEVEL) -gt 16 ] ; then echo -n 'y'; fi),y)
 EXTRA_CFLAGS += -DCONFIG_OMNIBOOK_BACKLIGHT
-else
-$(warning "Backlight support in only supported for kernel version newer than 2.6.16")
-$(warning "Disabling backlight sysfs interface")
-endif
 endif
 endif
 
diff -ur omnibook-9999.orig/polling.c omnibook-9999/polling.c
--- omnibook-9999.orig/polling.c	2011-10-28 23:33:07.000000000 +0600
+++ omnibook-9999/polling.c	2011-10-28 23:36:10.000000000 +0600
@@ -125,7 +125,7 @@
 	if(!key_polling_enabled)
 		goto out;
 
-	cancel_rearming_delayed_workqueue(omnibook_wq, &omnibook_poll_work);
+	cancel_delayed_work_sync(&omnibook_poll_work);
 	dprintk("Scancode emulation for volume buttons disabled.\n");
 	key_polling_enabled = 0;
 
@@ -188,7 +188,7 @@
 {
 	mutex_lock(&poll_mutex);
 	if(key_polling_enabled)
-		cancel_rearming_delayed_workqueue(omnibook_wq, &omnibook_poll_work);
+		cancel_delayed_work_sync(&omnibook_poll_work);
 	mutex_unlock(&poll_mutex);
 	return 0;
 }
