Index: omnibook/trunk/laptop.h
===================================================================
--- omnibook/trunk/laptop.h	(revisione 288)
+++ omnibook/trunk/laptop.h	(copia locale)
@@ -637,6 +637,15 @@
 	},
 	{
 		.callback = dmi_matched,
+		.ident = "Toshiba Satellite A300",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Satellite A300"),
+		},
+		.driver_data = (void*) TSA105
+	},
+	{
+		.callback = dmi_matched,
 		.ident = "Toshiba Satellite P100",
 		.matches = {
 			DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
Index: omnibook/trunk/acpi.c
===================================================================
--- omnibook/trunk/acpi.c	(revisione 288)
+++ omnibook/trunk/acpi.c	(copia locale)
@@ -66,6 +66,7 @@
 static char ec_dev_list[][20] = {
 	"\\_SB.PCI0.LPCB.EC0",
 	"\\_SB.PCI0.LPC0.EC0",
+	"\\_SB_.PCI0.LPC_.EC0",
 };
 
 /* TSX205 HCI and display handles */
