Index: acpi.c
===================================================================
--- acpi.c	(Revision 280)
+++ acpi.c	(Arbeitskopie)
@@ -683,11 +683,11 @@
 {
 	int retval;
 	int raw_state = !!(state & WIFI_STA);
+	u32 in[HCI_WORDS] = { HCI_SET, HCI_RF_CONTROL, raw_state, HCI_WIRELESS_POWER, 0, 0 };
+	u32 out[HCI_WORDS];
 
 	dprintk("set_wireless raw_state: %x\n", raw_state);
 
-	u32 in[HCI_WORDS] = { HCI_SET, HCI_RF_CONTROL, raw_state, HCI_WIRELESS_POWER, 0, 0 };
-	u32 out[HCI_WORDS];
 	hci_raw(in, out);
 
 	raw_state |= !!(state & BT_STA) << 0x1;	/* bit 1 */
@@ -842,11 +842,13 @@
 {
 	u32 in[HCI_WORDS] = { 0, 0, 0, 0, 0, 0 };
 	u32 out[HCI_WORDS];
+	acpi_status status;
 	in[0] = HCI_SET;
 	in[1] = HCI_HOTKEY_EVENT;
 	in[2] = (state & HKEY_FN) ? 1 : 0;
-	acpi_status status = hci_raw(in, out);
 
+	status = hci_raw(in, out);
+
 	dprintk("set_hotkeys (Fn interface) raw_state: %x\n", in[2]);
 
 	return (status == AE_OK) ? out[0] : HCI_FAILURE;
