--- a/src/groupchatdlg.cpp	2010-02-13 22:30:30.000000000 +0500
+++ b/src/groupchatdlg.cpp	2010-02-13 22:32:12.000000000 +0500
@@ -590,6 +590,13 @@
 					suggestedNicks << nick + postAdd;
 				}
 			}
+
+			foreach(QString link, mLinks) {
+				if (link.left(toComplete_.length()).lower() == toComplete_.lower()) {
+					suggestedNicks << link;
+				}
+			}
+
 			return suggestedNicks;
 		};
 
@@ -618,7 +625,25 @@
 			return p_->dlg->ui_.lv_users->nickList();
 		}
 
+		virtual void appendLink(const QString &message) {
+			QRegExp rx("<a href=\"(.+)\">");
+			int pos = rx.indexIn(message);
+			// no matches	
+			if (pos == -1) {
+				return;
+			}	
+			QStringList links = rx.capturedTexts();
+			for (int i = 1; i < links.size(); i += 2) {
+				// i += 2: every first capture is entire matching string, skip it
+				QString link = links.value(i, "");
+				if (link.isEmpty() || mLinks.contains(link)) // avoid empty and duplicates values
+					continue;
+				mLinks.prepend(link); // prepend is used because recent links should go first
+			}
+		}
+
 		QStringList mCmdList_;
+		QStringList mLinks;
 		
 		// FIXME where to move this?
 		QString nickSeparator; // equals ":"
@@ -1597,6 +1622,7 @@
 		txt = TextUtil::plain2rich(m.body());
 
 	txt = TextUtil::linkify(txt);
+	d->tabCompletion.appendLink(txt);
 
 	if(PsiOptions::instance()->getOption("options.ui.emoticons.use-emoticons").toBool())
 		txt = TextUtil::emoticonify(txt);