diff -bru psi-0.13/src/mainwin_p.cpp psi-0.13-meins/src/mainwin_p.cpp
--- psi/src/mainwin_p.cpp	2009-07-26 23:02:02.000000000 +0200
+++ psi-meins/src/mainwin_p.cpp	2010-01-13 00:33:42.000000000 +0100
@@ -35,6 +35,8 @@
 #include <QMenu>
 #include <QMouseEvent>
 #include <QtAlgorithms>
+#include <QProcess>
+#include <QStringList>
 
 #include "psiaccount.h"
 #include "stretchwidget.h"
@@ -42,6 +44,7 @@
 #include "icontoolbutton.h"
 #include "alerticon.h"
 #include "psicontactlist.h"
+#include "psioptions.h"
 
 //----------------------------------------------------------------------------
 // PopupActionButton
@@ -558,6 +561,15 @@
 	Q3PtrList<MLabel> labels;
 	bool hide;
 	QString message;
+	void executeExternal(bool show) { 
+		QString prog = PsiOptions::instance()->getOption("options.ui.notifications.external.external-notificator").toString();
+		QStringList args;
+		if (show) 
+			args+="1";
+		else
+			args+="0";
+		QProcess::startDetached(prog,args);
+	}
 };
 
 EventNotifierAction::EventNotifierAction(QObject *parent, const char *name)
@@ -621,6 +633,7 @@
 
 void EventNotifierAction::hide()
 {
+	if (!d->hide) d->executeExternal(false);
 	d->hide = true;
 
 	Q3PtrListIterator<MLabel> it(d->labels);
@@ -646,6 +659,7 @@
 
 void EventNotifierAction::show()
 {
+	if (d->hide) d->executeExternal(true);
 	d->hide = false;
 
 	Q3PtrListIterator<MLabel> it ( d->labels );
