--- a/src/psicontactlistviewdelegate.cpp	2010-02-13 22:37:24.000000000 +0500
+++ b/src/psicontactlistviewdelegate.cpp	2010-02-13 22:37:31.000000000 +0500
@@ -35,6 +35,10 @@
 	, font_(0)
 	, fontMetrics_(0)
 {
+	QColor bgColor = PsiOptions::instance()->getOption("options.ui.look.colors.contactlist.background").value<QColor>();
+	QPalette p = parent->palette();
+	p.setColor(QPalette::Base, bgColor);
+	parent->setPalette(p);
 	connect(PsiOptions::instance(), SIGNAL(optionChanged(const QString&)), SLOT(optionChanged(const QString&)));
 	optionChanged(contactListFontOptionPath);
 	optionChanged(showStatusMessagesOptionPath);
@@ -99,7 +103,7 @@
 
 	r.setLeft(avatarRect.right() + 3);
 
-	QColor textColor = Qt::black;
+	QColor textColor = PsiOptions::instance()->getOption("options.ui.look.colors.contactlist.status.online").value<QColor>();
 	if (statusType(index) == XMPP::Status::Away || statusType(index) == XMPP::Status::XA)
 		textColor = PsiOptions::instance()->getOption("options.ui.look.colors.contactlist.status.away").value<QColor>();
 	else if (statusType(index) == XMPP::Status::DND)