diff --git a/man/sfcbrepos.1.pre.in b/man/sfcbrepos.1.pre.in
index aa910c2..a3f7d2f 100644
--- a/man/sfcbrepos.1.pre.in
+++ b/man/sfcbrepos.1.pre.in
@@ -26,7 +26,7 @@ Supported command line options are:
 .TP
 \fB\-c\fR \fIschemadir\fR
 Path to obtain the CIM Schema classes.
-Default is \fI@datadir@/sfcb\fR
+Default is \fI@datadir@/mof/cim-current\fR
 .TP
 \fB\-s\fR \fIstagingdir\fR
 Path to sfcb staging area containing class MOFs and registration files
@@ -65,7 +65,7 @@ Default is \fIauto\fR, which will auto-detect based on the contents of the
 Display usage information and exit.
 .SH FILES
 .TP
-\fI@datadir@/sfcb/CIM/CIM_Schema.mof\fR
+\fI@datadir@/mof/cim-current/CIM/CIM_Schema.mof\fR
 CIM Schema
 .TP
 \fI@localstatedir@/lib/sfcb/registration/providerRegister\fR
diff --git a/sfcbrepos.sh.in b/sfcbrepos.sh.in
index 71a6c38..3965320 100644
--- a/sfcbrepos.sh.in
+++ b/sfcbrepos.sh.in
@@ -59,7 +59,7 @@ then
     echo -e "\t-X create repository in non-native format as specifed by argument"
     echo -e "\t-s specify staging directory [@localstatedir@/lib/sfcb/stage]"
     echo -e "\t-r specify repository directory [@localstatedir@/lib/sfcb/registration]"
-    echo -e "\t-c specify directory containing CIM Schema MOFs [@datadir@/sfcb/CIM]"
+    echo -e "\t-c specify directory containing CIM Schema MOFs [@datadir@/mof/cim-current/CIM]"
     echo -e "\t-t create tiny class repository by omitting inheritance information"
     echo -e "\t-z compress repository with gzip"
     echo
@@ -99,7 +99,7 @@ fi
 
 if [ -z "$cimschemadir" ]
 then
-    cimschemadir=${DESTDIR}@datadir@/sfcb/CIM
+    cimschemadir=${DESTDIR}@datadir@/mof/cim-current/CIM
 fi
 
 if [ -d $stagingdir ] && [ -f $stagingdir/default.reg ] &&
