--- cmake_modules/CommuniConfig.cmake.orig	2012-04-11 18:55:52.000000000 +0600
+++ cmake_modules/CommuniConfig.cmake	2012-04-11 18:56:42.000000000 +0600
@@ -1,5 +1,8 @@
-FIND_LIBRARY(IRC_LIBRARY NAMES Communi)
-FIND_PATH(IRC_INCLUDE_DIR NAMES "ircglobal.h" PATH_SUFFIXES Communi qt4/Communi )
+find_package(Qt4 REQUIRED)
+include( ${QT_USE_FILE} )
+
+FIND_LIBRARY(IRC_LIBRARY NAMES Communi PATHS ${QT_LIBRARY_DIR})
+FIND_PATH(IRC_INCLUDE_DIR NAMES "ircglobal.h" PATHS ${QT_INCLUDE_DIR} PATH_SUFFIXES Communi)
 
 # message( STATUS ${IRC_LIBRARY})
 if( IRC_LIBRARY AND IRC_INCLUDE_DIR )
