--- setup.py.orig	2010-01-13 20:33:21.078255736 +0200
+++ setup.py	2010-01-13 20:35:31.089757827 +0200
@@ -29,7 +29,7 @@
 import zlib
 
 OPENCL_INC_DIRS = []
-for path in ('/usr/local/opencl/OpenCL/common/inc','/opt/opencl/OpenCL/common/inc'):
+for path in ('/usr/local/opencl/OpenCL/common/inc','/opt/opencl/OpenCL/common/inc','/usr/include/CL/','/usr/include/'):
     if os.path.exists(path):
         OPENCL_INC_DIRS.append(path)
         break
