diff --git a/ipa/freeipa-1.2.2/ipa-client/ipa-install/ipa-client-install b/ipa/freeipa-1.2.2/ipa-client/ipa-install/ipa-client-install
--- a/ipa/freeipa-1.2.2/ipa-client/ipa-install/ipa-client-install
+++ b/ipa/freeipa-1.2.2/ipa-client/ipa-install/ipa-client-install
@@ -109,7 +109,7 @@
         sys.exit(1)
 
     try:
-        run(["/sbin/service", "nscd", "restart"])
+        run(["/sbin/rc-service", "nscd", "restart"])
     except:
         print "Failed to restart start the NSCD daemon"
 
@@ -263,6 +263,7 @@
         opts.append({'name':'nss_srv_domain', 'type':'option', 'value':cli_domain})
 
     opts.append({'name':'empty', 'type':'empty'})
+    # Gentoo require port 
     try:
         fstore.backup_file("/etc/ldap.conf")
         ldapconf.newConf("/etc/ldap.conf", opts)
@@ -361,7 +362,7 @@
 
     #Activate Name Server Caching Daemon
     try:
-        run(["/sbin/service", "nscd", "restart"])
+        run(["/sbin/rc-service", "nscd", "restart"])
     except:
         print "Failed to start the NSCD daemon"
         print "Caching of users/groups will not be available"
@@ -369,6 +370,7 @@
 
     try:
         run(["/sbin/chkconfig", "nscd", "on"])
+	# Gentoo require port ^^
     except:
         print "Failed to configure automatic startup of the NSCD daemon"
         print "Caching of users/groups will not be available after reboot"
diff --git a/ipa/freeipa-1.2.2/ipa-client/ipaclient/ntpconf.py b/ipa/freeipa-1.2.2/ipa-client/ipaclient/ntpconf.py
--- a/ipa/freeipa-1.2.2/ipa-client/ipaclient/ntpconf.py
+++ b/ipa/freeipa-1.2.2/ipa-client/ipaclient/ntpconf.py
@@ -47,14 +47,14 @@
 
 # Undisciplined Local Clock. This is a fake driver intended for backup
 # and when no outside source of synchronized time is available. 
-server	127.127.1.0	# local clock
+#server	127.127.1.0	# local clock
 #fudge	127.127.1.0 stratum 10	
 
 # Drift file.  Put this in a directory which the daemon can write to.
 # No symbolic links allowed, either, since the daemon updates the file
 # by creating a temporary in the same directory and then rename()'ing
 # it to the file.
-driftfile /var/lib/ntp/drift
+driftfile /var/lib/ntp/ntp.drift
 
 # Key file containing the keys and key identifiers used when operating
 # with symmetric key cryptography. 
@@ -71,13 +71,9 @@
 """
 
 ntp_sysconfig = """# Drop root to id 'ntp:ntp' by default.
-OPTIONS="-x -u ntp:ntp -p /var/run/ntpd.pid"
+# freeipa config this file - not Gentoo original config
 
-# Set to 'yes' to sync hw clock after successful ntpdate
-SYNC_HWCLOCK=yes
-
-# Additional options for ntpdate
-NTPDATE_OPTIONS=""
+NTPD_OPTS="-x -u ntp:ntp -p /var/run/ntpd.pid"
 """
 
 def config_ntp(server_fqdn, fstore = None):
@@ -96,16 +92,16 @@
     fd.close()
 
     if fstore:
-        fstore.backup_file("/etc/sysconfig/ntpd")
+        fstore.backup_file("/etc/conf.d/ntpd")
     else:
-        shutil.copy("/etc/sysconfig/ntpd", "/etc/sysconfig/ntpd.ipasave")
+        shutil.copy("/etc/conf.d/ntpd", "/etc/conf.d/ntpd.ipasave")
 
-    fd = open("/etc/sysconfig/ntpd", "w")
+    fd = open("/etc/conf.d/ntpd", "w")
     fd.write(ntp_sysconfig)
     fd.close()
 
     # Set the ntpd to start on boot
-    run(["/sbin/chkconfig", "ntpd", "on"])
+    run(["/sbin/rc-upadate", "add", "ntpd"])
 
     # Restart ntpd
-    run(["/sbin/service", "ntpd", "restart"])
+    run(["/sbin/rc-service", "ntpd", "restart"])
