--- configure.in.bak	2009-11-16 22:16:04.151912423 +1300
+++ configure.in	2009-11-16 22:17:13.640127394 +1300
@@ -232,7 +232,7 @@
 
 AC_MSG_CHECKING(for Python)
 PYTHON_LIBS=[$(python -c "from distutils import sysconfig
-print '/'.join(sysconfig.get_config_vars('LIBPL','LIBRARY'))+' '+sysconfig.get_config_var('LIBS')+' '+sysconfig.get_config_var('LINKFORSHARED')")]
+print '/'.join(sysconfig.get_config_vars('LIBDIR','LIBRARY'))+' '+sysconfig.get_config_var('LIBS')+' '+sysconfig.get_config_var('LINKFORSHARED')")]
 PYTHON_CFLAGS=[$(python -c "from distutils import sysconfig
 print sysconfig.get_python_inc()")]
 PYTHON_CFLAGS="-I$PYTHON_CFLAGS"
