diff --git a/GNUmakefile b/GNUmakefile
index 95b4b28..4a700ea 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -153,9 +153,7 @@ oolite_C_FILES = \
     legacy_random.c \
     strlcpy.c \
     OOTCPStreamDecoder.c \
-    OOPlanetData.c \
-	ioapi.c \
-	unzip.c
+    OOPlanetData.c
 
 
 OOLITE_DEBUG_FILES = \
diff --git a/GNUmakefile.postamble b/GNUmakefile.postamble
index 5d97459..f09ca5b 100644
--- a/GNUmakefile.postamble
+++ b/GNUmakefile.postamble
@@ -4,10 +4,6 @@
 # to live (Contents/Resources)
 #
 CP_FLAGS                         = -rf
-
-ifeq ($(debug),yes)
-    EXT                          =.dbg
-endif
                                  
 ifeq ($(findstring -gnu,$(GNUSTEP_HOST_OS)),-gnu)
     CP_FLAGS                     += -u
diff --git a/Makefile b/Makefile
index 5d93c57..8af1d91 100644
--- a/Makefile
+++ b/Makefile
@@ -65,7 +65,7 @@ debug: $(DEPS_DBG)
 
 .PHONY: release
 release: $(DEPS)
-	$(MAKE) -f GNUmakefile debug=no strip=yes
+	$(MAKE) -f GNUmakefile debug=no strip=no
 	mkdir -p AddOns && rm -rf AddOns/Basic-debug.oxp && cp -rf DebugOXP/Debug.oxp AddOns/Basic-debug.oxp
 
 .PHONY: release-deployment
