diff -ur a/src/Core/OOCPUInfo.m b/src/Core/OOCPUInfo.m
--- a/src/Core/OOCPUInfo.m	2012-02-23 18:55:32.000000000 +0600
+++ b/src/Core/OOCPUInfo.m	2012-02-23 18:57:30.000000000 +0600
@@ -28,7 +28,9 @@
 #if OOLITE_MAC_OS_X
 #include <sys/sysctl.h>
 #elif (OOLITE_LINUX || OOLITE_WINDOWS)
+#define __block __glibc_block
 #include <unistd.h>
+#undef __block
 #endif
 
 
diff -ur a/src/Core/Scripting/OOJSEngineTimeManagement.m b/src/Core/Scripting/OOJSEngineTimeManagement.m
--- a/src/Core/Scripting/OOJSEngineTimeManagement.m	2012-02-23 18:55:32.000000000 +0600
+++ b/src/Core/Scripting/OOJSEngineTimeManagement.m	2012-02-23 18:58:26.000000000 +0600
@@ -31,7 +31,9 @@
 #import "OOJSScript.h"
 #import "OOCollectionExtractors.h"
 #import "OOLoggingExtended.h"
+#define __block __glibc_block
 #include <unistd.h>
+#undef __block
 
 
 #if OO_DEBUG
