--- GNUmakefile.postamble.orig	2011-03-26 22:43:40.561323731 +0500
+++ GNUmakefile.postamble	2011-03-26 22:44:51.512436717 +0500
@@ -5,9 +5,6 @@
 #
 CP_FLAGS                         = -rf
 
-ifeq ($(debug),yes)
-    EXT                          =.dbg
-endif
                                  
 ifeq ($(findstring -gnu,$(GNUSTEP_HOST_OS)),-gnu)
     CP_FLAGS                     += -u
@@ -39,9 +36,6 @@
 endif
 	$(CP) $(CP_FLAGS) src/Cocoa/Info-Oolite.plist $(PROGDIR)/Resources/Info-gnustep.plist
 	$(CP) $(CP_FLAGS) $(GNUSTEP_OBJ_DIR_NAME)/$(SRC_BIN) $(PROGDIR)/$(DEST_BIN)
-ifeq ($(debug),no)
-	$(STRIP) $(PROGDIR)/$(DEST_BIN)
-endif
 ifeq ($(GNUSTEP_HOST_OS),mingw32)
     ifeq ($(GNUSTEP_HOST_CPU),x86_64)
                 $(CP) $(CP_FLAGS) deps/Windows-deps/x86_64/DLLs/*.dll $(PROGDIR)
--- GNUmakefile.orig	2014-12-27 01:47:00.718556710 +0500
+++ GNUmakefile	2014-12-27 02:01:34.797497722 +0500
@@ -149,9 +149,7 @@
     legacy_random.c \
     strlcpy.c \
     OOTCPStreamDecoder.c \
-    OOPlanetData.c \
-	ioapi.c \
-	unzip.c
+    OOPlanetData.c
 
 
 OOLITE_DEBUG_FILES = \
