diff -r 94175b222a79 configure.ac
--- a/configure.ac	Tue Oct 19 20:06:17 2010 +0300
+++ b/configure.ac	Tue Oct 19 22:02:59 2010 +0300
@@ -171,46 +171,53 @@
 AC_SUBST([LIBXML2_LIBS])
 
 dnl Check for OCaml (optional, for OCaml bindings).
-AC_PROG_OCAML
-AC_PROG_FINDLIB
-AM_CONDITIONAL([HAVE_OCAML],[test "x$OCAMLC" != "xno" -a "x$OCAMLFIND" != "xno"])
+dnl AC_ARG_WITH([ocaml]
+dnl	[AS_HELP_STRING([--with-ocaml],
 
-if test "x$OCAMLC" != "xno"; then
-    dnl Check if we have caml/unixsupport.h header (OCaml bindings only).
-    old_CFLAGS="$CFLAGS"
-    CFLAGS="$CFLAGS -I$OCAMLLIB"
-    AC_CHECK_HEADERS([caml/unixsupport.h],[],[],
-    [
-    #include <caml/config.h>
-    #include <caml/mlvalues.h>
-    ])
-    CFLAGS="$old_CFLAGS"
+dnl		[Ocaml bindings @<:@default=check@:>@])],
+dnl	 [
+	 AC_PROG_OCAML
+	 AC_PROG_FINDLIB
+dnl	 if test "x$OCAMLC" != "xno"; then
+dnl	 	old_CFLAGS="$CFLAGS"
+dnl		AC_CHECK_HEADERS([caml/unixsupport.h],[],[],
+dnl		[
+dnl		#include <caml/config.h>
+dnl		#include <caml/mlvalues.h>
+dnl		])
+dnl		CFLAGS="$old_CFLAGS"
+		
+		dnl Do we have function caml_raise_with_args?
+dnl	   	 f=caml_raise_with_args
+dnl       		 AC_MSG_CHECKING([for function $f])
+dnl 		echo "char $f (); char foo() { return $f (); }" > conftest.c
+dnl    			 rm -f conftest_ml.ml
+dnl			touch conftest_ml.ml
+dnl		     if $OCAMLOPT -c conftest.c 2>/dev/null && \
+dnl				$OCAMLOPT -c conftest_ml.ml 2>/dev/null && \
+dnl			$OCAMLOPT conftest.o conftest_ml.cmx -o conftest 2>/dev/null ; then
+dnl          			AC_DEFINE([HAVE_CAML_RAISE_WITH_ARGS],[1],
+dnl		     		[Defined if function caml_raise_with_args exists.])
+dnl        			AC_MSG_RESULT([found])
+dnl		    else
+dnl            			AC_MSG_RESULT([not found])
+dnl		    fi
+dnl	    	rm -f conftest conftest.* conftest_ml.*
+dnl	   fi
+dnl	   with_ocaml=yes
+dnl	 ],
+dnl	  [with_ocaml=no])
 
-    dnl Do we have function caml_raise_with_args?
-    f=caml_raise_with_args
-    AC_MSG_CHECKING([for function $f])
-    echo "char $f (); char foo() { return $f (); }" > conftest.c
-    rm -f conftest_ml.ml
-    touch conftest_ml.ml
-    if $OCAMLOPT -c conftest.c 2>/dev/null && \
-       $OCAMLOPT -c conftest_ml.ml 2>/dev/null && \
-       $OCAMLOPT conftest.o conftest_ml.cmx -o conftest 2>/dev/null ; then
-        AC_DEFINE([HAVE_CAML_RAISE_WITH_ARGS],[1],
-                  [Defined if function caml_raise_with_args exists.])
-        AC_MSG_RESULT([found])
-    else
-        AC_MSG_RESULT([not found])
-    fi
-    rm -f conftest conftest.* conftest_ml.*
-fi
+
+dnl AM_CONDITIONAL([HAVE_OCAML],[test "x$with_ocaml" != xno])
 
 dnl Check for Perl (optional, for Perl bindings).
 dnl XXX This isn't quite right, we should check for Perl devel library.
-AC_CHECK_PROG([PERL],[perl],[perl],[no])
+dnlAC_CHECK_PROG([PERL],[perl],[perl],[no])
 
 dnl Check for Perl modules that must be present to compile and
 dnl test the Perl bindings.
-missing_perl_modules=no
+dnlmissing_perl_modules=yes
 for pm in Test::More ExtUtils::MakeMaker IO::Stringy; do
     AC_MSG_CHECKING([for $pm])
     if ! perl -M$pm -e1 >/dev/null 2>&1; then
@@ -224,8 +231,8 @@
     AC_MSG_WARN([some Perl modules required to compile or test the Perl bindings are missing])
 fi
 
-AM_CONDITIONAL([HAVE_PERL],
-    [test "x$PERL" != "xno" -a "x$missing_perl_modules" != "xyes"])
+dnlAM_CONDITIONAL([HAVE_PERL],
+dnl    [test "x$PERL" != "xno" -a "x$missing_perl_modules" != "xyes"])
 
 dnl dnl Check for Python (optional, for Python bindings).
 dnl AC_CHECK_PROG([PYTHON],[python],[python],[no])
