--- a/src/Makefile.am	2015-03-22 15:10:30.685976674 +0500
+++ b/src/Makefile.am	2015-03-22 15:13:33.590968982 +0500
@@ -471,11 +471,11 @@
 
 yuv2rgb.fs.h: yuv2rgb.fs
 	cd $(abs_srcdir) && \
-	xxd -i yuv2rgb.fs | sed s/}\;/,0x00}\;/ > $(abs_builddir)/yuv2rgb.fs.h
+	xxdi.pl yuv2rgb.fs | sed s/}\;/0x00}\;/ > $(abs_builddir)/yuv2rgb.fs.h
 
 yuv2rgb.vs.h: yuv2rgb.vs
 	cd $(abs_srcdir) && \
-	xxd -i yuv2rgb.vs | sed s/}\;/,0x00}\;/ > $(abs_builddir)/yuv2rgb.vs.h
+	xxdi.pl yuv2rgb.vs | sed s/}\;/0x00}\;/ > $(abs_builddir)/yuv2rgb.vs.h
 
 #because make bundle serahc in this dir
 install-data-local:
