Index: security/jss/lib/config.mk
===================================================================
RCS file: /cvsroot/mozilla/security/jss/lib/config.mk,v
retrieving revision 1.19
diff -u -r1.19 config.mk
--- mozilla/security/jss/lib/config.mk	25 Apr 2004 15:02:19 -0000	1.19
+++ mozilla/security/jss/lib/config.mk	30 Jan 2007 21:08:06 -0000
@@ -72,13 +72,7 @@
 else
 
 EXTRA_SHARED_LIBS += \
-    -L$(DIST)/lib \
-    -lnss3 \
-    -lsmime3 \
-    -lssl3 \
-    -lplc4 \
-    -lplds4 \
-    -lnspr4 \
+    `pkg-config --libs nss` \
     $(JAVA_LIBS) \
     $(NULL)
 
