diff -r b5285c3e0f08 Makefile.am
--- a/Makefile.am	Sun Dec 04 13:40:32 2011 +0200
+++ b/Makefile.am	Sun Dec 04 13:43:37 2011 +0200
@@ -13,7 +13,7 @@
 PYTHON_CPPFLAGS = -I/usr/include/python$(PYTHON_VERSION)
 
 ## Targets
-SUBDIRS = po docs
+SUBDIRS = po 
 TESTS = tests/config_test.sh tests/files_test tests/pwhash_test tests/utils_test
 if LDAP
 TESTS += tests/default_pw_test tests/ldap_test
diff -r b5285c3e0f08 configure.in
--- a/configure.in	Sun Dec 04 13:40:32 2011 +0200
+++ b/configure.in	Sun Dec 04 13:43:37 2011 +0200
@@ -148,6 +148,5 @@
 dnl Does not even compile
 AM_CONDITIONAL([KRB5], [false])
 
-AC_CONFIG_FILES([Makefile po/Makefile.in docs/Makefile docs/reference/Makefile
-libuser.pc])
+AC_CONFIG_FILES([Makefile po/Makefile.in libuser.pc])
 AC_OUTPUT
