diff -uNr supertux-0.3.3_orig/src/addon/addon_manager.cpp supertux-0.3.3/src/addon/addon_manager.cpp
--- supertux-0.3.3_orig/src/addon/addon_manager.cpp     2010-03-01 18:40:58.000000000 +0600
+++ supertux-0.3.3/src/addon/addon_manager.cpp  2011-09-15 10:57:13.485913271 +0700
@@ -29,7 +29,6 @@
 #ifdef HAVE_LIBCURL
 #  include <curl/curl.h>
 #  include <curl/easy.h>
-#  include <curl/types.h>
 #endif
 
 #include "addon/addon.hpp"