=== modified file 'Makefile'
--- Makefile	2009-12-19 18:50:20 +0000
+++ Makefile	2010-03-15 19:02:36 +0000
@@ -15,6 +15,7 @@
 AWNPREFIX = $(PREFIX)
 CAIRODOCKPREFIX = $(PREFIX)
 INSTALL_LOG = install.log
+LIBDIR = $(PREFIX)/lib
 
 .PHONY : install
 .PHONY : uninstall
@@ -26,13 +27,13 @@
 # install
 install:
 
-	-install -d $(DESTDIR)/etc/gnomenu $(DESTDIR)$(PREFIX)/bin/ $(DESTDIR)$(PREFIX)/lib \
-	$(DESTDIR)$(PREFIX)/share $(DESTDIR)$(PREFIX)/lib/bonobo/servers $(DESTDIR)$(CAIRODOCKPREFIX)/share/cairo-dock/plug-ins/Dbus/third-party/GnoMenu
+	-install -d $(DESTDIR)/etc/gnomenu $(DESTDIR)$(PREFIX)/bin/ $(DESTDIR)$(LIBDIR) \
+	$(DESTDIR)$(PREFIX)/share $(DESTDIR)$(LIBDIR)/bonobo/servers $(DESTDIR)$(CAIRODOCKPREFIX)/share/cairo-dock/plug-ins/Dbus/third-party/GnoMenu
 	@echo $(PREFIX) > $(DESTDIR)/etc/gnomenu/prefix
 	python -u setup.py
 	
 	#-install src/bin/GnoMenu.py $(DESTDIR)$(PREFIX)/bin/
-	-cp -r src/lib/gnomenu $(DESTDIR)$(PREFIX)/lib
+	-cp -r src/lib/gnomenu $(DESTDIR)$(LIBDIR)
 	-cp -r src/share/gnomenu $(DESTDIR)$(PREFIX)/share
 	-cp -r src/share/avant-window-navigator $(DESTDIR)$(AWNPREFIX)/share
 	#-cp -r src/share/xfce4 $(DESTDIR)$(PREFIX)/share
@@ -40,20 +41,20 @@
 	-install src/share/cairo-dock/third-party/GnoMenu/* $(DESTDIR)$(CAIRODOCKPREFIX)/share/cairo-dock/plug-ins/Dbus/third-party/GnoMenu/
 	#-cp -r src/share/cairo-dock ~/.config/
 	-install src/bin/GnoMenu.py $(DESTDIR)$(PREFIX)/bin/
-	-install src/lib/bonobo/GNOME_GnoMenu.server $(DESTDIR)$(PREFIX)/lib/bonobo/servers
+	-install src/lib/bonobo/GNOME_GnoMenu.server $(DESTDIR)$(LIBDIR)/bonobo/servers
 	@echo "Makefile: GnoMenu installed."
 
 
 # uninstall
 uninstall:
 
-	rm -rf $(PREFIX)/lib/gnomenu
+	rm -rf $(LIBDIR)/gnomenu
 	rm -rf $(PREFIX)/share/gnomenu
 	rm -rf $(PREFIX)/share/avant-window-navigator/applets/GnoMenu.desktop
 	rm -rf $(PREFIX)/share/avant-window-navigator/applets/GnoMenu
 	#rm -rf $(PREFIX)/share/xfce4/panel-plugins/GnoMenu.desktop
 	rm -rf $(PREFIX)/bin/GnoMenu.py
-	rm -rf $(PREFIX)/lib/bonobo/servers/GNOME_GnoMenu.server
+	rm -rf $(LIBDIR)/bonobo/servers/GNOME_GnoMenu.server
 	rm -rf /etc/gnomenu/prefix
 	rm -rf ~/.gnomenu
 	rm -rf ~/.config/cairo-dock/third-party/GnoMenu

