diff --git a/gtk/propertybox.c b/gtk/propertybox.c
index 2b1e12c..9e35f3a 100644
--- a/gtk/propertybox.c
+++ b/gtk/propertybox.c
@@ -42,6 +42,7 @@ void linphone_gtk_fill_combo_box(GtkWidget *combo, const char **devices, const c
 		This dummy text needs to be removed first*/
 	}
 
+	if (p)
 	for(;*p!=NULL;++p){
 		if ( cap==CAP_IGNORE
 			|| (cap==CAP_CAPTURE && linphone_core_sound_device_can_capture(linphone_gtk_get_core(),*p))
