--- src/parser.hpp	2008-10-04 15:58:41.000000000 +0600
+++ src/parser.hpp.new	2008-10-04 15:58:04.000000000 +0600
@@ -25,6 +25,7 @@
 #  include "config.h"
 #endif
 
+#include <memory>
 #include <string>
 #include <map>
 #include <vector>
--- src/langs_2to3.hpp	2008-10-04 16:00:23.000000000 +0600
+++ src/langs_2to3.hpp.new	2008-10-04 16:00:14.000000000 +0600
@@ -1,6 +1,7 @@
 #ifndef _LANGS_2TO3_HPP_
 #define _LANGS_2TO3_HPP_
 
+#include <string>
 const int langs_2to3_count=185;
 extern std::pair<std::string, std::string> langs_2to3[langs_2to3_count];
 
