--- configure.in.orig	2010-06-26 15:35:50.360774878 +0300
+++ configure.in	2010-06-26 15:39:44.696760302 +0300
@@ -96,7 +96,7 @@
 
 
 test "x$prefix" = x. || test "x$prefix" = xNONE && prefix=/usr/local
-AC_DEFINE_UNQUOTED(CONFIG_FILE_PATH, "`eval echo "${prefix}/etc/sagan.conf"`", [Sagan configuration file]) 
+AC_DEFINE_UNQUOTED(CONFIG_FILE_PATH, "`eval echo "${eprefix}${sysconfdir}/sagan.conf"`", [Sagan configuration file]) 
 
 AC_DEFINE_UNQUOTED(PACKAGE_NAME, "sagan" )
 AC_DEFINE_UNQUOTED(PACKAGE_STRING, "Sagan $VERSION")
