--- configure.in.orig	2011-08-07 00:52:12.000000000 +0300
+++ configure.in	2011-08-07 00:52:59.000000000 +0300
@@ -7,7 +7,7 @@
 AC_INIT(libpki, 0.6.5, [libpki-users@lists.sourceforge.net], [libpki])
 AC_PREREQ(2.13)
 AC_CONFIG_AUX_DIR(build)
-AC_CONFIG_MACRO_DIR([m4])
+dnl AC_CONFIG_MACRO_DIR([m4])
 
 dnl Release Subversion (for binary packages)
 RC=1
@@ -21,7 +21,7 @@
 AC_SUBST(TODAY)
 
 mybits=""
-if [[ `uname -p` = "x86_64" ]] ; then
+if [[ `uname -m` = "x86_64" ]] ; then
 	mybits="64";
 fi
 AC_SUBST(mybits)
