--- extensions/xt_SYSRQ.c	2009-01-13 14:28:28.000000000 -0600
+++ extensions/xt_SYSRQ.c.new	2009-01-13 17:12:08.000000000 -0600
@@ -225,12 +225,13 @@
 	iph  = ipv6_hdr(skb);
 	udph = udp_hdr(skb);
 	len  = ntohs(udph->len) - sizeof(struct udphdr);
-
+#if 0
 	if (sysrq_debug)
 		printk(KERN_INFO KBUILD_MODNAME
 		       ": " NIP6_FMT ":%hu -> :%hu len=%u\n",
 		       NIP6(iph->saddr), ntohs(udph->source),
 		       ntohs(udph->dest), len);
+#endif
 	return sysrq_tg(udph + sizeof(struct udphdr), len);
 }
 
