--- src/cvs2cl.h.orig	2011-10-07 14:53:59.000000000 +0600
+++ src/cvs2cl.h	2011-10-07 14:54:30.000000000 +0600
@@ -25,6 +25,7 @@
 #include <sstream>
 
 #ifdef HAVE_LIBTINYXML
+#define TIXML_USE_STL
 #include <tinyxml.h>
 #else
 #include "tinyxml/tinyxml.h"
--- ./src/svn.h.orig	2011-10-07 14:52:56.000000000 +0600
+++ ./src/svn.h	2011-10-07 14:53:27.000000000 +0600
@@ -25,6 +25,7 @@
 #include <sstream>
 
 #ifdef HAVE_LIBTINYXML
+#define TIXML_USE_STL
 #include <tinyxml.h>
 #else
 #include "tinyxml/tinyxml.h"
