diff -ur a/src/virtManager/domain.py b/src/virtManager/domain.py
--- a/src/virtManager/domain.py	2011-03-24 15:39:28.000000000 -0500
+++ b/src/virtManager/domain.py	2011-06-07 11:33:59.000000000 -0500
@@ -765,8 +765,12 @@
 
             pcentCpuTime = (
                 (cpuTime) * 100.0 /
-                (((now - prevTimestamp) * 1000.0 * 1000.0 * 1000.0) *
-                 self.connection.host_active_processor_count()))
+                (((now - prevTimestamp) * 1000.0 * 1000.0 * 1000.0) ))
+
+            if self.vcpu_count() > 0:
+                 pcentCpuTime /= self.vcpu_count()
+            else:
+                 pcentCpuTime /= self.connection.host_active_processor_count()
 
             # Due to timing diffs between getting wall time & getting
             # the domain's time, its possible to go a tiny bit over
Only in b/src/virtManager: domain.py.orig
