autotools: install *.py in $(pkglibdir), ie $prefix/lib/gajim

diff -r 2c6ac74dd0e2 data/Makefile.am
--- a/data/Makefile.am	Tue Feb 03 15:52:16 2009 +0100
+++ b/data/Makefile.am	Tue Feb 03 16:52:41 2009 +0100
@@ -5,7 +5,7 @@
 desktop_in_files = gajim.desktop.in.in 
 desktop_DATA = $(desktop_in_files:.desktop.in.in=.desktop)
 
-installdefsdir = $(pkgdatadir)/src/common
+installdefsdir = $(pkglibdir)/common
 installdefs_DATA = defs.py
 
 soundsdir = $(pkgdatadir)/data/sounds
diff -r 2c6ac74dd0e2 scripts/gajim.in
--- a/scripts/gajim.in	Tue Feb 03 15:52:16 2009 +0100
+++ b/scripts/gajim.in	Tue Feb 03 16:52:41 2009 +0100
@@ -29,6 +29,6 @@
 export datadir=@DATADIR@/gajim
 PYTHON_EXEC=@PYTHON@
 
-cd ${datadir}/src
+cd @LIBDIR@/gajim
 export PYTHONPATH="$PYTHONPATH:@LIBDIR@/gajim"
 exec ${PYTHON_EXEC} -OO $APP.py "$@"
--- a/src/Makefile.am	2009-11-12 19:42:41.059691851 +0500
+++ b/src/Makefile.am	2009-11-12 19:43:52.670777307 +0500
@@ -26,26 +26,26 @@
 		    --override $(srcdir)/trayicon.override \
 		    $(srcdir)/trayicon.defs > $@
 endif
-gajimsrcdir = $(pkgdatadir)/src
+gajimsrcdir = $(pkgdatadir)
 gajimsrc_PYTHON = $(srcdir)/*.py
 
-gajimsrc1dir = $(pkgdatadir)/src/common
+gajimsrc1dir = $(pkgdatadir)/common
 gajimsrc1_PYTHON = \
 				$(srcdir)/common/*.py
 
-gajimsrc2dir = $(pkgdatadir)/src/common/xmpp
+gajimsrc2dir = $(pkgdatadir)/common/xmpp
 gajimsrc2_PYTHON = \
 				$(srcdir)/common/xmpp/*.py
 
-gajimsrc3dir = $(pkgdatadir)/src/common/zeroconf
+gajimsrc3dir = $(pkgdatadir)/common/zeroconf
 gajimsrc3_PYTHON = \
 				$(srcdir)/common/zeroconf/*.py
 
-gajimsrc4dir = $(pkgdatadir)/src/command_system
+gajimsrc4dir = $(pkgdatadir)/command_system
 gajimsrc4_PYTHON = \
 				   $(srcdir)/command_system/*.py
 
-gajimsrc5dir = $(pkgdatadir)/src/command_system/implementation
+gajimsrc5dir = $(pkgdatadir)/command_system/implementation
 gajimsrc5_PYTHON = \
 				   $(srcdir)/command_system/implementation/*.py
 
