From 0a89be24748cce94f3fa261f7c9494c7e942833b Mon Sep 17 00:00:00 2001
From: Kandrashin Denis <mail@lintest.ru>
Date: Mon, 28 Nov 2011 20:17:02 +0000
Subject: CR3: patch for libpng-1.5

---
diff --git a/3rdparty/crengine/src/lvimg.cpp b/3rdparty/crengine/src/lvimg.cpp
index e4b6412..b4f1fe8 100644
--- a/3rdparty/crengine/src/lvimg.cpp
+++ b/3rdparty/crengine/src/lvimg.cpp
@@ -657,7 +657,7 @@ bool LVPngImageSource::Decode( LVImageDecoderCallback * callback )
     if ( !png_ptr )
         return false;
 
-    if (setjmp( png_ptr->jmpbuf )) {
+    if (setjmp(png_jmpbuf(png_ptr))) {
         _width = 0;
         _height = 0;
         if (png_ptr)
@@ -676,7 +676,7 @@ bool LVPngImageSource::Decode( LVImageDecoderCallback * callback )
     if (!info_ptr)
         lvpng_error_func(png_ptr, "cannot create png info struct");
     png_set_read_fn(png_ptr,
-        (voidp)this, lvpng_read_func);
+        (void*)this, lvpng_read_func);
     png_read_info( png_ptr, info_ptr );
 
 
--
cgit v0.9.0.2
