From 999f00f4b0c9781d35ea7c9b3fa0f727510ecb49 Mon Sep 17 00:00:00 2001
From: Kandrashin Denis <mail@lintest.ru>
Date: Thu, 30 Jun 2011 12:26:18 +0000
Subject: fix syslog

---
diff --git a/sources/MyRuLib/FbLogStream.cpp b/sources/MyRuLib/FbLogStream.cpp
index 67e4e8f..4a737fd 100644
--- a/sources/MyRuLib/FbLogStream.cpp
+++ b/sources/MyRuLib/FbLogStream.cpp
@@ -41,7 +41,7 @@ bool FbLog::Update(wxArrayString &lines)
 
 #ifdef FB_SYSLOG_LOGGING
 
-#include "../version.inc"
+#include "config.h"
 
 extern "C" {
 #include <syslog.h>
@@ -49,7 +49,7 @@ extern "C" {
 
 FbLogSyslog::FbLogSyslog()
 {
-	openlog(wxSTRINGIZE(PROGRAM_NAME), LOG_PID, LOG_USER);
+	openlog(PACKAGE_NAME, LOG_PID, LOG_USER);
 }
 
 FbLogSyslog::~FbLogSyslog()
--
cgit v0.9.0.1
