diff --git a/src/Makefile b/src/Makefile
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,15 +1,10 @@
 # libdkim makefile for UNIX
 #
 
-#ifdef LINUX
-CFLAGS  = -c
-LFLAGS  = 
+DESTDIR ?= /
+LIBDIR ?= lib64
+CFLAGS  += -fPIC
 LIBS    = -lcrypto -lresolv
-#else
-CFLAGS  = -c
-LFLAGS  = 
-LIBS    = -lcrypto 
-#endif
 
 INCL	=  -I /usr/include/openssl/
 
@@ -19,23 +14,35 @@
 
 HDRS = dkim.h dns.h dkimbase.h dkimsign.h dkimverify.h
 
-all: libdkim.a libdkimtest
+all: libdkim.so.1 libdkimtest
 
 libdkim.a: $(OBJS)
 	rm -f libdkim.a
 	ar cr libdkim.a $(OBJS)
 	ranlib libdkim.a
 
-libdkimtest : libdkim.a libdkimtest.o
-	g++ -olibdkimtest $(LFLAGS) \
-	       -L . libdkimtest.o $(LIBS) -ldkim
+libdkim.so.1: $(OBJS)
+	$(CXX) -shared -Wl,-soname,$@.1 $(LDFLAGS) $(OBJS) -o $@ $(LIBS)
+	ln -s $@ libdkim.so
+
+.cpp.o:
+	$(CXX) $(CFLAGS) -c $<
+
+libdkimtest : libdkim.so.1 libdkimtest.o
+	$(CXX) -o$@ $(LDFLAGS) \
+	       -L. libdkimtest.o $(LIBS) -ldkim
 	
 libdkimtest.o: libdkimtest.cpp $(HDRS)
 
-.cpp.o:
-	g++ $(CFLAGS) -c $<
 
 clean:
-	rm  *.o libdkim.lib libdkimtest
+	rm -f *.o libdkim.{a,so,so.1} libdkimtest
 
+install: all
+	install -D -m 0755 libdkim.so.1 $(DESTDIR)/usr/$(LIBDIR)/libdkim.so.1
+	ln -s libdkim.so.1 $(DESTDIR)/usr/$(LIBDIR)/libdkim.so
+	ln -s libdkim.so.1 $(DESTDIR)/usr/$(LIBDIR)/libdkim.so.1.1
+	install -D -m 0644 dkim.h $(DESTDIR)/usr/include/dkim.h
+	install -D -m 0755 libdkimtest $(DESTDIR)/usr/bin/libdkimtest
 
+
