diff -up qpid-0.12/cpp/src/qpid/log/Logger.cpp\~ qpid-0.12/cpp/src/qpid/log/Logger.cpp
--- qpid-0.12/cpp/src/qpid/log/Logger.cpp~	2011-05-02 15:20:36.000000000 +0200
+++ qpid-0.12/cpp/src/qpid/log/Logger.cpp	2011-12-07 21:11:15.238671000 +0100
@@ -23,7 +23,7 @@
 #include "qpid/sys/Thread.h"
 #include "qpid/sys/Time.h"
 #include "qpid/DisableExceptionLogging.h"
-#include <boost/pool/detail/singleton.hpp>
+#include <boost/serialization/singleton.hpp>
 #include <boost/bind.hpp>
 #include <boost/function.hpp>
 #include <algorithm>
@@ -45,7 +45,7 @@ inline void Logger::enable_unlocked(Stat
 }
 
 Logger& Logger::instance() {
-    return boost::details::pool::singleton_default<Logger>::instance();
+    return boost::serialization::singleton<Logger>::get_mutable_instance();
 }
 
 Logger::Logger() : flags(0) {

