--- xdg/Menu.py	2009-10-18 22:24:03.306293446 +0600
+++ xdg/Menu.py	2009-10-18 22:23:23.854335273 +0600
@@ -499,6 +499,10 @@
         filename = __getFileName(filename)
 
     # use default if no filename given
+    if not filename:
+        prefix = "XDG_MENU_PREFIX" in os.environ and os.environ["XDG_MENU_PREFIX"] or ""
+        filename = __getFileName(prefix + "applications.menu")
+
     if not filename:
         filename = __getFileName("applications.menu")
 
