--- configure.ac.orig	2009-04-12 19:30:17.010337388 +0400
+++ configure.ac	2009-04-12 19:30:48.347564072 +0400
@@ -18,7 +18,7 @@
 PKG_CHECK_MODULES(libusb, libusb >= 0.1.12)
 PKG_CHECK_MODULES(libglib2, glib-2.0 >= 2.14.1)
 PKG_CHECK_MODULES(libgthread2, gthread-2.0 >= 2.14.1)
-PKG_CHECK_MODULES(libgnutls, gnutls >= 1.6.3 gnutls <= 2.5.0 )
+PKG_CHECK_MODULES(libgnutls, gnutls >= 1.6.3)
 PKG_CHECK_MODULES(libtasn1, libtasn1 >= 1.1)
 
 # Checks for header files.
