diff -r d76126b76399 sipwitch-0.7.4/configure.ac
--- a/sipwitch-0.7.4/configure.ac	Tue Mar 16 13:48:49 2010 +0200
+++ b/sipwitch-0.7.4/configure.ac	Tue Mar 16 14:15:27 2010 +0200
@@ -354,14 +354,14 @@
 AC_DEFINE_UNQUOTED(DEFAULT_LIBPATH, "$default_libpath", [lib path])
 AC_DEFINE_UNQUOTED(DEFAULT_LIBEXEC, "$default_libexec", [script path])
 
-zeroconf=""
-ZEROCONF_LIBS=""
-if test -z "$zeroconf" ; then
-    PKG_CHECK_MODULES(ZEROCONF, avahi-client > 0.3, [
-        AC_DEFINE(ZEROCONF_AVAHI, 1, [zeroconf avahi client])
-        zeroconf="avahi"
-    ],[zeroconf="$zeroconf"])
-fi
+#zeroconf=""
+#ZEROCONF_LIBS=""
+#if test -z "$zeroconf" ; then
+#    PKG_CHECK_MODULES(ZEROCONF, avahi-client > 0.3, [
+#        AC_DEFINE(ZEROCONF_AVAHI, 1, [zeroconf avahi client])
+#        zeroconf="avahi"
+#    ],[zeroconf="$zeroconf"])
+#fi
 
 AH_BOTTOM([
 #ifndef	HAVE_LOCALTIME_R
