--- configure.ac.orig	2011-05-16 00:06:58.000000000 +0300
+++ configure.ac	2011-05-16 00:10:03.000000000 +0300
@@ -342,13 +342,16 @@
 
 zeroconf=""
 ZEROCONF_LIBS=""
-if test -z "$zeroconf" ; then
+AC_ARG_ENABLE(zeroconf,
+    AC_HELP_STRING([--with-zeroconf],[enable zeroconf support]),[
+
+if test  "$enable_zeroconf" != "xNO" ; then
     PKG_CHECK_MODULES(ZEROCONF, avahi-client > 0.3, [
         AC_DEFINE(ZEROCONF_AVAHI, 1, [zeroconf avahi client])
         zeroconf="avahi"
     ],[zeroconf="$zeroconf"])
 fi
-
+])
 AH_BOTTOM([
 #ifndef HAVE_LOCALTIME_R
 #define localtime_r(x, y)   localtime(x)
